src/controllers/incident.js

```diff
@@ -0,0 +1,180 @@
+              * cos( radians( humans.latitude ) )
+              * cos( radians( humans.longitude ) - radians(${data.longitude}) )
+              + sin( radians(${data.latitude}) )
+              * sin( radians( humans.latitude ) ) ) < incident.distance and incident.distance != 0) or
+			  incident.distance = 0 and (${areastring}))
+               group by humans.id, humans.name, incident.template`
+			log.log({ level: 'debug', message: 'incidentWhoCares query', event: 'sql:incidentWhoCares' })
+			this.db.query(query)
+				.then((result) => {
+					log.info(`Incident on ${data.name} appeared and ${result[0].length} humans cared`)
+					resolve(result[0])
+				})
+				.catch((err) => {
+					log.error(`incidentWhoCares errored with: ${err}`)
+				})
+		})
+	}
+
+	async handle(data) {
+		return new Promise((resolve) => {
+			switch (config.geocoding.staticProvider.toLowerCase()) {
+				case 'google': {
+					data.staticmap = `https://maps.googleapis.com/maps/api/staticmap?center=${data.latitude},${data.longitude}&markers=color:red|${data.latitude},${data.longitude}&maptype=${config.geocoding.type}&zoom=${config.geocoding.zoom}&size=${config.geocoding.width}x${config.geocoding.height}&key=${_.sample(config.geocoding.staticKey)}`
+					break
+				}
+				case 'osm': {
+					data.staticmap = `https://www.mapquestapi.com/staticmap/v5/map?locations=${data.latitude},${data.longitude}&size=${config.geocoding.width},${config.geocoding.height}&defaultMarker=marker-md-3B5998-22407F&zoom=${config.geocoding.zoom}&key=${_.sample(config.geocoding.staticKey)}`
+					break
+				}
+				case 'mapbox': {
+					data.staticmap = `https://api.mapbox.com/styles/v1/mapbox/streets-v10/static/url-https%3A%2F%2Fi.imgur.com%2FMK4NUzI.png(${data.longitude},${data.latitude})/${data.longitude},${data.latitude},${config.geocoding.zoom},0,0/${config.geocoding.width}x${config.geocoding.height}?access_token=${_.sample(config.geocoding.staticKey)}`
+					break
+				}
+				default: {
+					data.staticmap = ''
+				}
+			}
+            if(data.name) data.gym_name = data.name;
+
+			data.mapurl = `https://www.google.com/maps/search/?api=1&query=${data.latitude},${data.longitude}`
+			data.applemap = `https://maps.apple.com/maps?daddr=${data.latitude},${data.longitude}`
+			
+			let incidentExpiration = data.incident_expiration ? data.incident_expiration : data.incident_expire_timestamp;
+			data.tth = moment.preciseDiff(Date.now(), incidentExpiration * 1000, true)
+			data.distime = moment(incidentExpiration * 1000).tz(geoTz(data.latitude, data.longitude).toString()).format(config.locale.time)
+			
+			// Stop handling if it already disappeared or is about to go away
+			if (data.tth.firstDateWasLater || ((data.tth.hours * 3600) + (data.tth.minutes * 60) + data.tth.seconds) < minTth) {
+				log.warn(`${data.name} Incident already disappeared or is about to go away in: ${data.tth.hours}:${data.tth.minutes}:${data.tth.seconds}`)
+				resolve([])
+				return null
+			}
```

Comment:
We are not passing the TTH so we should not need this check. We can do that NOW() check in the other file like your TODO had.

Reply:
tth is calculated a few lines above, so this code should prevent any late discoveries, just like with pokemons.